x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.test.ts

```diff
@@ -204,5 +204,14 @@ describe('manifest_manager', () => {
         oldArtifactId
       );
     });
+
+    test('ManifestManager handles promise rejections when building artifacts', async () => {
+      // This test won't fail on an unhandled promise rejection, but it will cause
+      // an UnhandledPromiseRejectionWarning to be printed.
+      const manifestManager = getManifestManagerMock({
+        mockType: ManifestManagerMockType.ListClientPromiseRejection,
+      });
+      await expect(manifestManager.buildNewManifest()).rejects.toThrow();
+    });
```

Comment:
I think this test passes without the changes `manifest_manager.ts` https://github.com/elastic/kibana/pull/73831/files#diff-21a709e8a87929d93a7efdb95905d024R85-R95

Reply:
We zoomed. It passes either way, but an `UnhandledPromiseRejectionWarning` is printed when you run it against the pre-change code, and it's absent now. Doesn't appear to be a way to force jest to behave differently...

Reply:
@madirey explained to me, as posted in the comment, the test will pass without those changes, but the changes prevent the rejection warning from appearing when running the tests. The rejection does get handled further down the line when building the manifest.